controller/metrics/utils.py

```diff
@@ -20,13 +20,15 @@ def resource_request_from_manifest(manifest: Dict[str, Any]) -> Optional[Resourc
     resources = manifest.get("spec", {}).get("jupyterServer", {}).get("resources", {}).get(
         "requests", {}
     )
+    resources = {**resources}
```

Comment:
This is essentially a nop? or what does this do?

If it is just to cash to a plain python dict, `resources = dict(resources)` communicates the intent better, imo.

Reply:
@Panaetius yeah I essentially wanted a copy of that dictionary. Because in a few lines below I add something to it just for the purposes of this function. But if I do not use a copy then that same dictionary in the manifest gets modified in the same way and breaks things.

Now I just use `copy.deepcopy` for this which is the safest option and properly communicates the intent IMO.